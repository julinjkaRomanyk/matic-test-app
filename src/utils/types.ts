export type InfoStore = {
    firstName: string;
    lastName: string;
    phoneNumber: string;
    stateName: string;
};

export type Progress = {
    current: number;
    total: number;
};

export enum Fields {
    nameQuestion = "nameQuestion",
    phoneQuestion = "phoneQuestion",
    stateQuestion = "stateQuestion",
};


export type DefaultOption = { 
    label: string; 
    value: string 
};