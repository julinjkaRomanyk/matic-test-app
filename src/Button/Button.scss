.Button {
    height: 60px;
    width: 185px;
    border-radius: 30px;
    background-color: #ff5122;
    color: white;
    font-size: 20px;
    font-weight: bold;
    border: none;
}

.Button:hover {
    background-color: white;
    color: #ff5122;
    border: 2px solid #ff5122;
    cursor: pointer;
}

.Button:active {
    background-color: #ffbe1e;
    color: #ff5122;
    border: 2px solid #ff5122;
}

.Button:focus {
    outline: 0;
}
