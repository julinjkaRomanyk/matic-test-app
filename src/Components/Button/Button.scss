@import "utils/_variables";

.Button {
    height: 60px;
    width: 185px;
    border-radius: 30px;
    background-color: $BasicOrange;
    color: $BasicWhite;
    font-size: 20px;
    font-weight: bold;
    border: none;
}

.Button:hover {
    background-color: $BasicWhite;
    color: $BasicOrange;
    border: 2px solid $BasicOrange;
    cursor: pointer;
}

.Button:active {
    background-color: $SecondOrange;
    color: $BasicOrange;
    border: 2px solid$BasicOrange;
}

.Button:focus {
    outline: 0;
}
