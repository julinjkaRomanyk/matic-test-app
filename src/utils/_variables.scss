$BasicBlack: black;
$BasicOrange: #FF5122;
$BasicWhite: white;
$SecondOrange: #FFBE1E;
$BackgroundColor: #FCFAF9;
$Grey: #726969;
$BorderColor: #AFA7A6;
$DarkPowder: #F9ECE7;
