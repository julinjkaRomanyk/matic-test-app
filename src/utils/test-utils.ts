export function getTestName(dirname: string = __dirname) {
    return dirname.split("/").reverse()[0];
  };