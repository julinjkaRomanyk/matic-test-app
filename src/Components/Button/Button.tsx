import React from "react";

import "./Button.scss"

export type Props = {
    label?: string;
    onClick(): void;
}

export const Button: React.FC<Props> = ({ label, onClick }) => {
    return (
        <button className="Button" onClick={onClick} id="button" data-testid="button">
            {label}
        </button>
    )
}
