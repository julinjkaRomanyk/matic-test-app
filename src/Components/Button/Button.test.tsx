import React from "react";
import { getTestName } from "utils/test-utils";
import { render, fireEvent } from "@testing-library/react";

import { Button } from "./Button";

const fakeButtonCallback = jest.fn();

describe(getTestName(__dirname), () => {
    test("Match snapshot", () => {
      const { container } = render(<Button label="" onClick={fakeButtonCallback} />);
  
      expect(container.firstChild).toMatchSnapshot();
    });

    test("Should call fakeButtonCallback onClick", () => {
        const { getByTestId } = render(<Button label="" onClick={fakeButtonCallback} />);
    
        const button = getByTestId("button")
        fireEvent.click(button);

        expect(fakeButtonCallback).toBeCalledTimes(1);
      });

  });
  