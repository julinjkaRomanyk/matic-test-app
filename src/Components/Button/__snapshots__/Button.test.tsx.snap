// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Match snapshot 1`] = `
<button
  class="Button"
  data-testid="button"
  id="button"
/>
`;
